src/Agent.Listener/Environment.cs

```diff
@@ -0,0 +1,213 @@
+            //XPLAT is using process.platform, which returns 'darwin', 'freebsd', 'linux', 'sunos' or 'win32'
+            //windows agent is printing enviroment variable "OS", which is something like "Windows_NT" even when running Windows 10
+            //.Net core API RuntimeInformation.OSDescription is returning "Microsoft Windows 10.0.10586", 
+            //"Linux 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014", "Darwin 15.4.0 Darwin Kernel Version 15.4.0: Fri Feb 26 22:08:05 PST 2016;"
+            caps["Agent.OS"] = RuntimeInformation.OSDescription;
+            caps["Agent.ComputerName"] = System.Environment.MachineName;
+
+            return caps;
+        }
+
+        private void GetRegularCapabilities(Dictionary<string, string> capabilities, CancellationToken token)
+        {
+            foreach (var cap in _regularCapabilities)
+            {
+                string capPath = IOUtil.Which(cap.Tool ?? cap.Name);
+                if (!string.IsNullOrEmpty(capPath))
+                {
+                    capabilities.Add(cap.Name, capPath);
+                }
+
+                else if (cap.Paths != null)
+                {
+                    foreach (var path in cap.Paths)
+                    {
+                        if (File.Exists(path))
+                        {
+                            capabilities.Add(cap.Name, path);
+                            break;
+                        }
+                    }
+                }
+
+                token.ThrowIfCancellationRequested();
+            }
+        }
+
+        private async Task GetShellCapabilities(Dictionary<string, string> capabilities, CancellationToken token)
+        {
+            foreach (var cap in _shellCapabilities)
+            {
+                var toolPath = IOUtil.Which(cap.Command);
+                if (string.IsNullOrEmpty(toolPath))
+                {
+                    continue;
+                }
+
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    string toolOutput = string.Empty;
+                    var outputHandler = new EventHandler<DataReceivedEventArgs>((sender, args) =>
+                    {
+                        if (!string.IsNullOrEmpty(args.Data))
+                        {
+                            toolOutput += args.Data;
+                        }
+                    });
+                    processInvoker.OutputDataReceived += outputHandler;
+                    try
+                    {
+                        await processInvoker.ExecuteAsync(
+                                    workingDirectory: string.Empty,
+                                    fileName: toolPath,
+                                    arguments: cap.CommandArgs,
+                                    environment: null,
+                                    cancellationToken: token);
+                        if (!string.IsNullOrEmpty(toolOutput))
```

Comment:
I don't know. The XPLAT is not checking if the path returned by shell capability exists.